carma_wm/src/WMListenerWorker.cpp

```diff
@@ -52,8 +51,32 @@ void WMListenerWorker::mapCallback(const autoware_lanelet2_msgs::MapBinConstPtr&
     map_callback_();
   }
 }
-void WMListenerWorker::mapUpdateCallback(const autoware_lanelet2_msgs::MapBinConstPtr& geofence_msg) const
+
+bool WMListenerWorker::checkIfReRoutingNeeded() const
+{
+  return rerouting_flag_;
+}
+
+void WMListenerWorker::enableUpdatesWithoutRoute()
+{
+  route_node_flag_=true;
+}
+
+void WMListenerWorker::mapUpdateCallback(const autoware_lanelet2_msgs::MapBinConstPtr& geofence_msg)
 {
+
+  if(geofence_msg->invalidates_route==true)
+  {  
+    rerouting_flag_=true;
+    local_geofence_msg_ = boost::make_shared<autoware_lanelet2_msgs::MapBin>(*geofence_msg);
+
+    if(route_node_flag_!=true)
+    {
+     ROS_INFO_STREAM("Route is not invalidated/available yet");
```

Comment:
This alert should just say Route is not available yet. "Not invalidated" is a bit confusing since it was invalidated.